src/Identifier/Resolver/ResolverAwareTrait.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (http://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+
+namespace Authentication\Identifier\Resolver;
+
+use Cake\Core\App;
+use InvalidArgumentException;
+use RuntimeException;
+
+trait ResolverAwareTrait
```

Comment:
@burzum It's under `Identifier` namespace so it should be obvious what it's used for. 

In ORM we have a "generic" `LocatorAwareInterface` and `LocatorInterface` but it's under `Cake\ORM` so the scope is somewhat clear.